src/kubernetes.cc

```diff
@@ -455,12 +461,15 @@ std::vector<MetadataUpdater::ResourceMetadata>
   }
   std::vector<MetadataUpdater::ResourceMetadata> result;
 
-  const std::string node_name = CurrentNode();
+  const std::string current_node = CurrentNode();
 
   if (config_.VerboseLogging()) {
-    LOG(INFO) << "Current node is " << node_name;
+    LOG(INFO) << "Current node is " << current_node;
   }
 
+  const std::string node_name(
+      config_.KubernetesClusterLevelMetadata() ? "" : current_node);
```

Comment:
I'm concerned that the responses are very different here. When a node name is provided, a single node is returned. Is there any logic that distinguishes between a list / single node?

Reply:
Good catch. I forgot that in one case you get a `Node` and in another a `NodeList`. I've updated the code to be more robust.